packages/backend-common/src/reading/types.ts

```diff
@@ -22,6 +22,7 @@ import { Config } from '@backstage/config';
  */
 export type UrlReader = {
   read(url: string): Promise<Buffer>;
+  readTree(url: string): Promise<ReadTreeResponse>;
```

Comment:
Was thinking we could make this optional, but tbh I think the simplest and most clear way forward for now is to just throw an error with something like "File tree reading is not yet implemented for GitLab URLs" in all the readers that don't implement `readTree`.

Reply:
Already done, It's coming in a commit in a few minutes. Will just replace the klaw lib with the recursive-readdir lib and update the PR.